radio/src/boards/rm-h750/bsp_io.cpp

```diff
@@ -66,7 +66,7 @@ static uint32_t _read_io_expander(bsp_io_expander* io)
   if (pca95xx_read(&io->exp, io->mask, &value) == 0) {
     io->state = value;
   } else {
-    // First safety level: reset TCA chip
+    // Unable to read PCA, reset it
     TRACE("ERROR: resetting PCA95XX");
     gpio_clear(IO_RESET_GPIO);
     delay_us(1);  // Only 4ns are needed according to PCA datasheet
```

Comment:
No, it is a minimum of 6ns + min 400ns (at 3V3) for it to reset . But that shouldn't be an issue though, right, since this is a delay for 1000ns? Just double-checking as the delay is *not* present in bsp_io_init() toggling of reset ;)

<img width="698" height="144" alt="image" src="https://github.com/user-attachments/assets/20be84d3-5a92-44e6-b92d-920884877f4a" />

Reply:
Well pulse duration is what matters here imho, since thats what this delay mostly pilots. Possibly chip will be ready a bit later (while still under the 1us), and we could add a delay later, but since we don't configure or send to the chip anything, I think it doesn't matter. We could add a 1  us wait at init before bsp_io_read_switches, but current code doesn't produce any error.

Reply:
lol... you can ignore most of this now since you've changed it ;) Anyway, did you notice if the GX12 suffers from this lockup also? Is it worth adding there anyway?

Sort of feel the PCA driver is missing a reset routine. Meaning the other French guy didn't finish the job :-P 

tRESET is the delay of the IC responding _after_ the active low reset pin is set high again. If I'm reading the code right, the timeout for the read should be 10ms, thus well in excess of the tRESET period. But wouldn't that also mean we are _relying_ on that rather than have correct-ish timings? Just thinking it could lead to inconsistent behaviour if the I2C bus is reset *as well*. Just something to think about since still playing wack-a-mole with TCA9539 and GX12 switches.

Reply:
While this fix has been tested to successfully reset the bus on TX15 using the process described in the issue (both by me and RM), it does not seem work on GX12 based on users test. Still unsure why since I cannot replicate

Reply:
:cry: